rootfs/etc/init.d/ssh-audit

```diff
@@ -7,13 +7,13 @@ if [ "${SSH_AUDIT_ENABLED}" == "true" ]; then
 		Defaults!/usr/bin/sudoreplay !log_output
 		Defaults!REBOOT !log_output
 	EOF
-	echo "# valid login shells" > /etc/shells
 	echo "/usr/bin/sudosh" >> /etc/shells
 	echo "session requisite pam_exec.so quiet /usr/bin/sudosh-add-user" > /etc/pam.d/sudosh
+	echo "auth required pam_shells.so" > /etc/pam.d/chsh
 	[ -w /etc/passwd ] && chsh -s /usr/bin/sudosh root
+	echo "# valid login shells" > /etc/shells
+	echo "/usr/bin/sudosh" >> /etc/shells
```

Comment:
Why do we have this twice?

Reply:
@osterman sorry, it's been a while so my memory is a little fuzzy on this and I'm not an expert with pam/pam modules. I believe what I'm doing here is adding `chsh` to the authorized shells, then changing shells, then truncating the `/etc/shells` file with the same comment you had originally and then appending the `sudosh` shell. Otherwise you get a pam error when trying to `chsh` in the first place. It could probably be cleaner, but this got it working for me.